mainapp/views.py

```diff
@@ -223,29 +227,77 @@ class Meta:
        widgets = {
            'address': forms.Textarea(attrs={'rows':3}),
            'notes': forms.Textarea(attrs={'rows':3}),
-           'gender': forms.RadioSelect()
+           'gender': forms.RadioSelect(),
         }
 
 
     def __init__(self, *args, **kwargs):
-       user = kwargs.pop('user')
+       camp_id = kwargs.pop('camp_id')
        super(PersonForm, self).__init__(*args, **kwargs)
-       self.fields['camped_at'].queryset = RescueCamp.objects.filter(data_entry_user=user)
-       self.fields['camped_at'].initial = RescueCamp.objects.filter(data_entry_user=user).first()
+       self.fields['camped_at'].queryset = RescueCamp.objects.filter(id=camp_id)
+       self.fields['camped_at'].initial = RescueCamp.objects.filter(id=camp_id).first()
 
 class AddPerson(SuccessMessageMixin,LoginRequiredMixin,CreateView):
     login_url = '/login/'
     model = Person
     template_name='mainapp/add_person.html'  
     form_class = PersonForm
-    success_url = '/add_person/'
     success_message = "'%(name)s' registered successfully"
 
+    def get_success_url(self):
+        return reverse('add_person', args=(self.camp_id,))
+    def dispatch(self, request, *args, **kwargs):
+        self.camp_id = kwargs.get('camp_id','')
+        
+        try:
+            self.camp = RescueCamp.objects.get(id=int(self.camp_id))
+        except ObjectDoesNotExist:
+            raise Http404
+        if request.user!=self.camp.data_entry_user:
+            raise PermissionDenied
+        return super(AddPerson, self).dispatch(request, *args, **kwargs)
+
     def get_form_kwargs(self):
         kwargs = super(AddPerson, self).get_form_kwargs()
-        kwargs['user'] = self.request.user
+        kwargs['camp_id'] = self.camp_id
         return kwargs
 
+
+class CampDetailsForm(forms.ModelForm):
+    class Meta:
+       model = RescueCamp
+       fields = [
+        'name',
+        'food_req',
+        'clothing_req',
+        'sanitary_req',
+        'medical_req',
+        'other_req'
+        ]
+       read_only = ('name',)
+       widgets = {
+           'name': forms.Textarea(attrs={'rows':1,'readonly':True}),
+           'food_req': forms.Textarea(attrs={'rows':3}),
+           'clothing_req': forms.Textarea(attrs={'rows':3}),
+           'medical_req': forms.Textarea(attrs={'rows':3}),
+           'sanitary_req': forms.Textarea(attrs={'rows':3}),
+           'other_req': forms.Textarea(attrs={'rows':3}),
+       }
+
+class CampDetails(SuccessMessageMixin,LoginRequiredMixin,UpdateView):
+    login_url = '/login/'
+    model = RescueCamp
+    template_name='mainapp/camp_details.html'  
+    form_class = CampDetailsForm
+    success_url = '/coordinator_home/'
```

Comment:
`/` suffix needed ?

Reply:
context https://github.com/IEEEKeralaSection/rescuekerala/pull/294

Reply:
#294 says we need trailing / or use {% url %}